manga-finder/src/components/MangaReader.tsx

```diff
@@ -0,0 +1,52 @@
+import { ChaoticOrbit } from "@uiball/loaders";
+import { IChapter, IChapterResponse } from "../models/IChapter";
+import { getChapterPages } from "../services/PageQueries";
+
+function MangaReader() {
+  function getChapterID() {
+    const chapterID = window.location.pathname.substring(
+      7,
+      window.location.pathname.length
+    );
+    return chapterID;
+  }
+  const { isLoading, error, data } = getChapterPages(getChapterID());
+
+  if (isLoading)
+    return (
+      <div className="flex h-screen w-screen flex-col justify-center align-middle content-center items-center">
+        <h2 className="text-2xl">Loading</h2>
+        <ChaoticOrbit size={35} color="#FFFFFF" />
+      </div>
+    );
+
+  if (error) return <h1>An error has occured</h1>;
+
+  const response: IChapterResponse = data;
+
+  const chapter: IChapter = response.chapter;
+
+  const chapterPages: string[] = chapter.data;
+
+  let counter = 1;
+  return (
+    <div className="manga-reader bg-dark w-screen h-auto flex flex-col items-center">
+      {chapterPages.map((page) => {
+        return (
+          <>
+            <img
+              src={
+                "https://uploads.mangadex.org/data/" + chapter.hash + "/" + page
```

Comment:
Maybe try using the data-saver images to make loading quicker.

Reply:
Right, I'll do some last minute changes.

Reply:
Did the changes minus the data-saver, thought maybe people would like the quality despite the sometimes slow loading.